js_modules/dagster-ui/packages/ui-components/src/theme/theme.tsx

```diff
@@ -28,15 +29,22 @@ export const getTheme = memoize(() => {
   }
 
   // Allow setting the theme as an override.
-  if (value === DagsterTheme.Light || value === DagsterTheme.Dark) {
+  if (
+    value === DagsterTheme.Light ||
+    value === DagsterTheme.Dark ||
+    value === DagsterTheme.Legacy
+  ) {
     return value;
   }
 
-  // todo dish: Uncomment this to allow OS settings to work automatically.
-  // if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
-  //   return DagsterTheme.Dark;
-  // }
+  if (value === DagsterTheme.System) {
+    if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
```

Comment:
This looks good to me! I like that the memoize means this is only re-computed on page load so systems that change at dusk won't start returning `dark` as individual components on the page re-render.